Cypress rebase

Kelketek opened the pull request.

Kelketek commented:
@mattdrayer @antoviaque This is the rebase in progress. It's not done yet-- I'm still doing cleanup. But it's already shaping up to be a lot cleaner than last time around.

antoviaque commented:
@Kelketek Thanks! Glad to see this : )

Btw, for the review, can you make sure only "new" commit show up, ie commits that weren't on master before the rebase (either on the solutions fork master, or from cypress). The idea being to only show the modifications done to make the rebase work, not all the commits from the rebase itself - as that would be too big to review. It won't show the conflict resolutions you did within the rebase, but I don't know of a way to show this in a PR.

Note that from a quick look, the commits above are all from master before the rebase, right? Are there any significant changes you made in conflict resolution during the rebase that would be worth isolating for review?

Kelketek commented:
@antoviaque Thus far I've been fixing things by using conflict resolution-- that is, I've been editing history a bit to make previous changes work with the new code. I've been doing this because it will make the next rebase a lot simpler. I had to keep re-correcting things because of the 'paint on top' method we have been using. However after a few more changes in that manner, I can start doing that instead, if preferred.

As for what's particularly notable, that would be things like the grading system. We added a field 'due' while upstream added the field 'module_id'. I've added both in and modified the code to take both. I don't know how this might affect outside tools.
